app/components/instruction/ed25519/Ed25519DetailsCard.tsx

```diff
@@ -77,6 +84,178 @@ const extractData = (
     }
 };
 
+function decodeMessageFromAnchorProgram(
+    anchorProgram: Program,
+    message: Uint8Array
+): { name: string; data: any } | null {
+    const messageDisc = Buffer.from(message.slice(0, 8)).toString('hex');
+    const coder = anchorProgram.coder.types;
+    for (const [_, typeLayouts] of Object.entries(anchorProgram.coder.types)) {
+        for (const [name] of typeLayouts.entries()) {
+            try {
+                const capitalizedName = name.charAt(0).toUpperCase() + name.slice(1);
+                const disc = Buffer.from(sha256(`global:${capitalizedName}`).slice(0, 8)).toString('hex');
+
+                if (disc === messageDisc) {
+                    const decoded = coder.decode(name, Buffer.from(message.slice(8)));
+                    if (decoded) {
+                        return { data: decoded, name };
+                    }
+                }
+            } catch (e) {
+                console.log('Error decoding message with anchor program', e);
+            }
+        }
+    }
+    return null;
+}
+
+function SignatureDetails({
+    index,
+    offset,
+    signature,
+    pubkey,
+    message,
+    messageIx,
+}: {
+    index: number;
+    offset: Ed25519SignatureOffsets;
+    signature: Uint8Array | null;
+    pubkey: Uint8Array | null;
+    message: Uint8Array;
+    messageIx: PartiallyDecodedInstruction;
+}) {
+    const { url } = useCluster();
+    const anchorProgram = useAnchorProgram(messageIx.programId.toBase58(), url);
+
+    const decodedMessage = useMemo(() => {
+        if (!anchorProgram?.idl || !anchorProgram?.program) {
+            return null;
+        }
+        return decodeMessageFromAnchorProgram(anchorProgram.program, Buffer.from(message.toString(), 'hex'));
+    }, [anchorProgram, message]);
+
+    const messageRow = useMemo(() => {
+        if (!decodedMessage || !anchorProgram?.idl || !anchorProgram?.program) {
+            return (
+                <tr>
+                    <td>Message</td>
+                    <td
+                        className="text-lg-end"
+                        style={{
+                            fontSize: '0.85rem',
+                            lineHeight: '1.2',
+                            maxWidth: '100%',
+                            overflowWrap: 'break-word',
+                            whiteSpace: 'normal',
+                            wordBreak: 'break-all',
+                        }}
+                    >
+                        <Copyable text={Buffer.from(message).toString('base64')}>
+                            <span className="font-monospace">{Buffer.from(message).toString('base64')}</span>
+                        </Copyable>
+                    </td>
+                </tr>
+            );
+        }
+
+        const name = decodedMessage.name;
+        const data = decodedMessage.data;
+
+        if (!name || !data) {
+            return null;
```

Comment:
Should you instead just fallback to rendering the message as a base64 string like in the clause above?

Reply:
yea you were right